test/e2e/framework.go

```diff
@@ -145,13 +150,32 @@ func (f *Framework) afterEach() {
 		Logf("Found DeleteNamespace=false, skipping namespace deletion!")
 	}
 
+	summaries := make([]TestDataSummary, 0)
 	if testContext.GatherKubeSystemResourceUsageData {
-		f.gatherer.stopAndPrintData([]int{50, 90, 99, 100}, f.addonResourceConstraints)
+		summaries = append(summaries, f.gatherer.stopAndSummarize([]int{50, 90, 99, 100}, f.addonResourceConstraints))
 	}
 
 	if testContext.GatherLogsSizes {
 		close(f.logsSizeCloseChannel)
 		f.logsSizeWaitGroup.Wait()
+		summaries = append(summaries, f.logsSizeVerifier.GetSummary())
+	}
+
+	outputTypes := strings.Split(testContext.OutputPrintType, ",")
+	Logf("Found output types: %v", outputTypes)
+	for _, printType := range outputTypes {
+		switch printType {
+		case "hr":
+			for i := range summaries {
+				Logf(summaries[i].PrintHumanReadable())
+			}
+		case "json":
+			for i := range summaries {
+				Logf(summaries[i].PrintJSON())
+			}
+		default:
```

Comment:
I think that structure:

for i := range summaries {
  switch printType {
    case "hr": Logf(summaries[i].PrintHumanReadable())
    case "json": Logf(summaries[i].PrintJSON())
    default: Logf(...)
  }
}

is more readable and shorter. WDYT?